internal/driver/testdata/testfixture.js

```diff
@@ -0,0 +1,46 @@
+// TestFixture records log messages and errors in an array that will
+// be returned to Go code. Each element in the result array is either
+// an array of the form ["LOG", ...]", or ["ERROR", ...].
+class TestFixture {
+  constructor() {
+    this.context = "";  // Added to front of all log and error messages.
+    this.result = [];
+  }
+
+  run(name, subtest) {
+    this.result.push(["LOG", "===", name]);
+    this.context = "";
+    subtest();
+    this.context = "";
+  }
+
+  // setContext arranges to add name(args) to all messages added
+  // until the next setContext or phase call.
+  setContext(name, ...args) {
+    let str = name + "(";
+    let sep = "";
+    for (const arg of args) {
+      str += sep;
+      str += arg;
+      sep = ",";
+    }
+    str += "):";
+    this.context = str;
+  }
+
+  log(...args) {
+    this.result.push(this.toArray("LOG", this.context, ...args));
```

Comment:
I don't know JS that well, but poking around a bit I think this can be just `this.result.push("LOG", this.context, ...args);`.

Reply:
Thanks. You were close. I needed:

```
this.result.push(["LOG", this.context, ...args]);
```